File: CustomApplications/VB.Net/HohmannTransferUsingTargeter/Form1.vb
Imports AGI.STKObjects
Imports AGI.STKObjects.Astrogator
Imports AGI.STKUtil

Public Class Form1

    Private stkRootObject As IAgStkObjectRoot = Nothing
    Private _driver As IAgVADriverMCS

    Private ReadOnly Property stkRoot() As AGI.STKObjects.IAgStkObjectRoot
        Get
            If stkRootObject Is Nothing Then
                Dim STKXApp As AGI.STKX.AgSTKXApplication = New AGI.STKX.AgSTKXApplication

                If STKXApp.IsFeatureAvailable(AGI.STKX.AgEFeatureCodes.eFeatureCodeGlobeControl) Then
                    stkRootObject = TryCast(New AgStkObjectRoot(), IAgStkObjectRoot)
                    If stkRootObject.AvailableFeatures.IsPropagatorTypeAvailable(AgEVePropagatorType.ePropagatorAstrogator) Then
                        stkRootObject = TryCast(New AgStkObjectRoot(), IAgStkObjectRoot)
                    Else
                        MessageBox.Show("You do not have the Astrogator license.", "License Error", MessageBoxButtons.OK, MessageBoxIcon.Stop)
                        Application.Exit()
                    End If
                Else
                    MessageBox.Show("You do not have the required license.", "License Error", MessageBoxButtons.OK, MessageBoxIcon.Stop)
                    Application.Exit()
                End If
            End If
            stkRoot = stkRootObject
        End Get
    End Property

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
       
    End Sub

    Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
        button1.Enabled = False
        stkRoot.NewScenario("HohmannTransfer")
        Dim sat1 As IAgSatellite = TryCast(stkRoot.CurrentScenario.Children.[New](AgESTKObjectType.eSatellite, "Satellite1"), IAgSatellite)
        sat1.SetPropagatorType(AgEVePropagatorType.ePropagatorAstrogator)
        _driver = TryCast(sat1.Propagator, IAgVADriverMCS)
        _driver.MainSequence.RemoveAll()
        button2.Enabled = True
    End Sub

    Private Sub button2_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button2.Click
        button2.Enabled = False
        Dim initState As IAgVAMCSInitialState = TryCast(_driver.MainSequence.Insert(AgEVASegmentType.eVASegmentTypeInitialState, "Inner Orbit", "-"), IAgVAMCSInitialState)
        initState.SetElementType(AgEVAElementType.eVAElementTypeKeplerian)
        Dim modKep As IAgVAElementKeplerian = TryCast(initState.Element, IAgVAElementKeplerian)
        modKep.PeriapsisRadiusSize = 6700
        modKep.ArgOfPeriapsis = 0
        modKep.Eccentricity = 0
        modKep.Inclination = 0
        modKep.RAAN = 0
        modKep.TrueAnomaly = 0
        button3.Enabled = True
    End Sub

    Private Sub button3_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button3.Click
        button3.Enabled = False
        Dim propagate As IAgVAMCSPropagate = TryCast(_driver.MainSequence.Insert(AgEVASegmentType.eVASegmentTypePropagate, "Propagate", "-"), IAgVAMCSPropagate)
        propagate.PropagatorName = "Earth Point Mass"
        DirectCast(propagate.StoppingConditions("Duration").Properties, IAgVAStoppingCondition).Trip = 7200
        button4.Enabled = True
    End Sub

    Private Sub button4_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button4.Click
        button4.Enabled = False
        Dim ts As IAgVAMCSTargetSequence = TryCast(_driver.MainSequence.Insert(AgEVASegmentType.eVASegmentTypeTargetSequence, "Start Transfer", "-"), IAgVAMCSTargetSequence)
        Dim dv1 As IAgVAMCSManeuver = TryCast(ts.Segments.Insert(AgEVASegmentType.eVASegmentTypeManeuver, "DV1", "-"), IAgVAMCSManeuver)
        dv1.SetManeuverType(AgEVAManeuverType.eVAManeuverTypeImpulsive)
        Dim impulsive As IAgVAManeuverImpulsive = TryCast(dv1.Maneuver, IAgVAManeuverImpulsive)
        impulsive.SetAttitudeControlType(AgEVAAttitudeControl.eVAAttitudeControlThrustVector)
        Dim thrustVector As IAgVAAttitudeControlImpulsiveThrustVector = TryCast(impulsive.AttitudeControl, IAgVAAttitudeControlImpulsiveThrustVector)
        thrustVector.ThrustAxesName = "Satellite/Satellite1 VNC(Earth)"
        dv1.EnableControlParameter(AgEVAControlManeuver.eVAControlManeuverImpulsiveCartesianX)
        DirectCast(dv1, IAgVAMCSSegment).Results.Add("Keplerian Elems/Radius of Apoapsis")
        Dim dc As IAgVAProfileDifferentialCorrector = TryCast(ts.Profiles("Differential Corrector"), IAgVAProfileDifferentialCorrector)
        Dim xControlParam As IAgVADCControl = dc.ControlParameters.GetControlByPaths("DV1", "ImpulsiveMnvr.Cartesian.X")
        xControlParam.Enable = True
        xControlParam.MaxStep = 0.3
        Dim roaEquality As IAgVADCResult = dc.Results.GetResultByPaths("DV1", "Radius Of Apoapsis")
        roaEquality.Enable = True
        roaEquality.DesiredValue = 42238
        roaEquality.Tolerance = 0.1
        dc.MaxIterations = 50
        dc.EnableDisplayStatus = True
        dc.Mode = AgEVAProfileMode.eVAProfileModeIterate
        ts.Action = AgEVATargetSeqAction.eVATargetSeqActionRunActiveProfiles
        button5.Enabled = True
    End Sub

    Private Sub button5_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button5.Click
        button5.Enabled = False
        Dim transferEllipse As IAgVAMCSPropagate = TryCast(_driver.MainSequence.Insert(AgEVASegmentType.eVASegmentTypePropagate, "Transfer Ellipse", "-"), IAgVAMCSPropagate)
        transferEllipse.PropagatorName = "Earth Point Mass"
        transferEllipse.StoppingConditions.Add("Apoapsis")
        transferEllipse.StoppingConditions.Remove("Duration")
        button6.Enabled = True
    End Sub

    Private Sub button6_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button6.Click
        button6.Enabled = False
        Dim ts As IAgVAMCSTargetSequence = TryCast(_driver.MainSequence.Insert(AgEVASegmentType.eVASegmentTypeTargetSequence, "Finish Transfer", "-"), IAgVAMCSTargetSequence)
        Dim dv2 As IAgVAMCSManeuver = TryCast(ts.Segments.Insert(AgEVASegmentType.eVASegmentTypeManeuver, "DV2", "-"), IAgVAMCSManeuver)
        dv2.SetManeuverType(AgEVAManeuverType.eVAManeuverTypeImpulsive)
        Dim impulsive As IAgVAManeuverImpulsive = TryCast(dv2.Maneuver, IAgVAManeuverImpulsive)
        impulsive.SetAttitudeControlType(AgEVAAttitudeControl.eVAAttitudeControlThrustVector)
        Dim thrustVector As IAgVAAttitudeControlImpulsiveThrustVector = TryCast(impulsive.AttitudeControl, IAgVAAttitudeControlImpulsiveThrustVector)
        thrustVector.ThrustAxesName = "Satellite/Satellite1 VNC(Earth)"
        dv2.EnableControlParameter(AgEVAControlManeuver.eVAControlManeuverImpulsiveCartesianX)
        DirectCast(dv2, IAgVAMCSSegment).Results.Add("Keplerian Elems/Eccentricity")
        Dim dc As IAgVAProfileDifferentialCorrector = TryCast(ts.Profiles("Differential Corrector"), IAgVAProfileDifferentialCorrector)
        Dim xControlParam As IAgVADCControl = dc.ControlParameters.GetControlByPaths("DV2", "ImpulsiveMnvr.Cartesian.X")
        xControlParam.Enable = True
        xControlParam.MaxStep = 0.3
        Dim eccConstraint As IAgVADCResult = dc.Results.GetResultByPaths("DV2", "Eccentricity")
        eccConstraint.Enable = True
        eccConstraint.DesiredValue = 0
        dc.EnableDisplayStatus = True
        dc.Mode = AgEVAProfileMode.eVAProfileModeIterate
        ts.Action = AgEVATargetSeqAction.eVATargetSeqActionRunActiveProfiles
        button7.Enabled = True
    End Sub

    Private Sub button7_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button7.Click
        button7.Enabled = False
        Dim outerOrbit As IAgVAMCSPropagate = TryCast(_driver.MainSequence.Insert(AgEVASegmentType.eVASegmentTypePropagate, "Outer Orbit", "-"), IAgVAMCSPropagate)
        outerOrbit.PropagatorName = "Earth Point Mass"
        DirectCast(outerOrbit.StoppingConditions("Duration").Properties, IAgVAStoppingCondition).Trip = 86400
        button8.Enabled = True
    End Sub

    Private Sub button8_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button8.Click
        Me.button8.Enabled = False        
        _driver.RunMCS()
        Dim startTransfer As IAgVAMCSTargetSequence = TryCast(_driver.MainSequence("Start Transfer"), IAgVAMCSTargetSequence)
        Dim finishTransfer As IAgVAMCSTargetSequence = TryCast(_driver.MainSequence("Finish Transfer"), IAgVAMCSTargetSequence)
        Dim startDC As IAgVAProfileDifferentialCorrector = TryCast(startTransfer.Profiles("Differential Corrector"), IAgVAProfileDifferentialCorrector)
        Console.WriteLine(startDC.ControlParameters.GetControlByPaths("DV1", "ImpulsiveMnvr.Cartesian.X").FinalValue)
        Dim finishDC As IAgVAProfileDifferentialCorrector = TryCast(finishTransfer.Profiles("Differential Corrector"), IAgVAProfileDifferentialCorrector)
        Console.WriteLine(finishDC.ControlParameters.GetControlByPaths("DV2", "ImpulsiveMnvr.Cartesian.X").FinalValue)
        Dim dv1 As IAgVAMCSManeuver = TryCast(startTransfer.Segments("DV1"), IAgVAMCSManeuver)
        Dim dv1Impulsive As IAgVAManeuverImpulsive = TryCast(dv1.Maneuver, IAgVAManeuverImpulsive)
        Dim dv1ThrustVector As IAgVAAttitudeControlImpulsiveThrustVector = TryCast(dv1Impulsive.AttitudeControl, IAgVAAttitudeControlImpulsiveThrustVector)
        Dim cartesian As IAgCartesian = TryCast(dv1ThrustVector.DeltaVVector.ConvertTo(AgEPositionType.eCartesian), IAgCartesian)
        Console.WriteLine(cartesian.X)
        startTransfer.ApplyProfiles()
        cartesian = TryCast(dv1ThrustVector.DeltaVVector.ConvertTo(AgEPositionType.eCartesian), IAgCartesian)
        Console.WriteLine(cartesian.X)

        Dim dv2 As IAgVAMCSManeuver = TryCast(finishTransfer.Segments("DV2"), IAgVAMCSManeuver)
        Dim dv2Impulsive As IAgVAManeuverImpulsive = TryCast(dv2.Maneuver, IAgVAManeuverImpulsive)
        Dim dv2ThrustVector As IAgVAAttitudeControlImpulsiveThrustVector = TryCast(dv2Impulsive.AttitudeControl, IAgVAAttitudeControlImpulsiveThrustVector)
        cartesian = TryCast(dv2ThrustVector.DeltaVVector.ConvertTo(AgEPositionType.eCartesian), IAgCartesian)
        Console.WriteLine(cartesian.X)
        finishTransfer.ApplyProfiles()
        cartesian = TryCast(dv2ThrustVector.DeltaVVector.ConvertTo(AgEPositionType.eCartesian), IAgCartesian)
        Console.WriteLine(cartesian.X)
        Me.button10.Enabled = True
    End Sub

    Private Sub Form1_FormClosing(ByVal sender As System.Object, ByVal e As System.Windows.Forms.FormClosingEventArgs) Handles MyBase.FormClosing
        If Not (IsNothing(stkRootObject)) Then
            stkRootObject.CloseScenario()
        End If
    End Sub

    Private Sub button10_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles button10.Click
        Dim dataForm As New Form2()
        Dim reportData As String = ""
        Dim outerOrbit As IAgVAMCSPropagate = TryCast(_driver.MainSequence("Outer Orbit"), IAgVAMCSPropagate)
        Dim result As IAgDrResult = DirectCast(outerOrbit, IAgVAMCSSegment).ExecSummary
        Console.WriteLine(result.Category)
        Dim intervals As IAgDrIntervalCollection = TryCast(result.Value, IAgDrIntervalCollection)
        For i As Integer = 0 To intervals.Count - 1
            Dim interval As IAgDrInterval = intervals(i)
            Dim datasets As IAgDrDataSetCollection = interval.DataSets
            For j As Integer = 0 To datasets.Count - 1
                Dim dataset As IAgDrDataSet = datasets(j)
                dataForm.FormTitle = dataset.ElementName
                Dim elements As System.Array = dataset.GetValues()
                For Each o As Object In elements
                    reportData += o.ToString() + "" & Chr(13) & "" & Chr(10) & ""
                Next
            Next
        Next
        dataForm.ReportData = reportData
        dataForm.ShowDialog()
    End Sub
End Class

File: CustomApplications/CSharp/HohmannTransferUsingTargeter/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using AGI.STKObjects;
using AGI.STKObjects.Astrogator;
using AGI.STKUtil;

namespace HohmannTransferUsingTargeter
{
    public partial class Form1 : Form
    {
        private IAgStkObjectRoot stkRootObject = null;
        private IAgVADriverMCS _driver;

        private AGI.STKObjects.IAgStkObjectRoot stkRoot
        {
            get
            {
                if (stkRootObject == null)
                {
                    stkRootObject = new AGI.STKObjects.AgStkObjectRootClass();
                }
                return stkRootObject;
            }
        }

        public Form1()
        {
            InitializeComponent();           
        }

        private void button1_Click(object sender, EventArgs e)
        {
            button1.Enabled = false;
            stkRoot.NewScenario("HohmannTransfer");
            IAgSatellite sat1 = stkRoot.CurrentScenario.Children.New(AgESTKObjectType.eSatellite, "Satellite1") as IAgSatellite;
            sat1.SetPropagatorType(AgEVePropagatorType.ePropagatorAstrogator);
            _driver = sat1.Propagator as IAgVADriverMCS;
            _driver.MainSequence.RemoveAll();
            _driver.Options.DrawTrajectoryIn3D = true;
            button2.Enabled = true;
        }

        private void button2_Click(object sender, EventArgs e)
        {
            button2.Enabled = false;
            IAgVAMCSInitialState initState = _driver.MainSequence.Insert(AgEVASegmentType.eVASegmentTypeInitialState, "Inner Orbit", "-") as IAgVAMCSInitialState;
            initState.SetElementType(AgEVAElementType.eVAElementTypeKeplerian);
            IAgVAElementKeplerian modKep = initState.Element as IAgVAElementKeplerian;
            modKep.PeriapsisRadiusSize = 6700;
            modKep.ArgOfPeriapsis = 0;
            modKep.Eccentricity = 0;
            modKep.Inclination = 0;
            modKep.RAAN = 0;
            modKep.TrueAnomaly = 0;
            button3.Enabled = true;
        }

        private void button3_Click(object sender, EventArgs e)
        {
            button3.Enabled = false;
            IAgVAMCSPropagate propagate = _driver.MainSequence.Insert(AgEVASegmentType.eVASegmentTypePropagate, "Propagate", "-") as IAgVAMCSPropagate;
            propagate.PropagatorName = "Earth Point Mass";
            ((IAgVAMCSSegment)propagate).Properties.Color = Color.Blue;
            ((IAgVAStoppingCondition)propagate.StoppingConditions["Duration"].Properties).Trip = 7200;
            button4.Enabled = true;
        }

        private void button4_Click(object sender, EventArgs e)
        {
            button4.Enabled = false;
            IAgVAMCSTargetSequence ts = _driver.MainSequence.Insert(AgEVASegmentType.eVASegmentTypeTargetSequence, "Start Transfer", "-") as IAgVAMCSTargetSequence;
            IAgVAMCSManeuver dv1 = ts.Segments.Insert(AgEVASegmentType.eVASegmentTypeManeuver, "DV1", "-") as IAgVAMCSManeuver;
            dv1.SetManeuverType(AgEVAManeuverType.eVAManeuverTypeImpulsive);
            IAgVAManeuverImpulsive impulsive = dv1.Maneuver as IAgVAManeuverImpulsive;
            impulsive.SetAttitudeControlType(AgEVAAttitudeControl.eVAAttitudeControlThrustVector);
            IAgVAAttitudeControlImpulsiveThrustVector thrustVector = impulsive.AttitudeControl as IAgVAAttitudeControlImpulsiveThrustVector;
            thrustVector.ThrustAxesName = "Satellite/Satellite1 VNC(Earth)";
            dv1.EnableControlParameter(AgEVAControlManeuver.eVAControlManeuverImpulsiveCartesianX);
            ((IAgVAMCSSegment)dv1).Results.Add("Keplerian Elems/Radius of Apoapsis");
            IAgVAProfileDifferentialCorrector dc = ts.Profiles["Differential Corrector"] as IAgVAProfileDifferentialCorrector;
            IAgVADCControl xControlParam = dc.ControlParameters.GetControlByPaths("DV1", "ImpulsiveMnvr.Cartesian.X");
            xControlParam.Enable = true;
            xControlParam.MaxStep = 0.3;
            IAgVADCResult roaResult = dc.Results.GetResultByPaths("DV1", "Radius Of Apoapsis");
            roaResult.Enable = true;
            roaResult.DesiredValue = 42238;
            roaResult.Tolerance = 0.1;
            dc.MaxIterations = 50;
            dc.EnableDisplayStatus = true;
            dc.Mode = AgEVAProfileMode.eVAProfileModeIterate;
            ts.Action = AgEVATargetSeqAction.eVATargetSeqActionRunActiveProfiles;
            button5.Enabled = true;
        }

        private void button5_Click(object sender, EventArgs e)
        {
            button5.Enabled = false;
            IAgVAMCSPropagate transferEllipse = _driver.MainSequence.Insert(AgEVASegmentType.eVASegmentTypePropagate, "Transfer Ellipse", "-") as IAgVAMCSPropagate;
            ((IAgVAMCSSegment)transferEllipse).Properties.Color = Color.Red;
            transferEllipse.PropagatorName = "Earth Point Mass";
            transferEllipse.StoppingConditions.Add("Apoapsis");
            transferEllipse.StoppingConditions.Remove("Duration");
            button6.Enabled = true;
        }

        private void button6_Click(object sender, EventArgs e)
        {
            button6.Enabled = false;
            IAgVAMCSTargetSequence ts = _driver.MainSequence.Insert(AgEVASegmentType.eVASegmentTypeTargetSequence, "Finish Transfer", "-") as IAgVAMCSTargetSequence;
            IAgVAMCSManeuver dv2 = ts.Segments.Insert(AgEVASegmentType.eVASegmentTypeManeuver, "DV2", "-") as IAgVAMCSManeuver;
            dv2.SetManeuverType(AgEVAManeuverType.eVAManeuverTypeImpulsive);
            IAgVAManeuverImpulsive impulsive = dv2.Maneuver as IAgVAManeuverImpulsive;
            impulsive.SetAttitudeControlType(AgEVAAttitudeControl.eVAAttitudeControlThrustVector);
            IAgVAAttitudeControlImpulsiveThrustVector thrustVector = impulsive.AttitudeControl as IAgVAAttitudeControlImpulsiveThrustVector;
            thrustVector.ThrustAxesName = "Satellite/Satellite1 VNC(Earth)";
            dv2.EnableControlParameter(AgEVAControlManeuver.eVAControlManeuverImpulsiveCartesianX);
            ((IAgVAMCSSegment)dv2).Results.Add("Keplerian Elems/Eccentricity");
            IAgVAProfileDifferentialCorrector dc = ts.Profiles["Differential Corrector"] as IAgVAProfileDifferentialCorrector;
            IAgVADCControl xControlParam = dc.ControlParameters.GetControlByPaths("DV2", "ImpulsiveMnvr.Cartesian.X");
            xControlParam.Enable = true;
            xControlParam.MaxStep = 0.3;
            IAgVADCResult eccResult = dc.Results.GetResultByPaths("DV2", "Eccentricity");
            eccResult.Enable = true;
            eccResult.DesiredValue = 0;
            dc.EnableDisplayStatus = true;
            dc.Mode = AgEVAProfileMode.eVAProfileModeIterate;
            ts.Action = AgEVATargetSeqAction.eVATargetSeqActionRunActiveProfiles;
            button7.Enabled = true;
        }

        private void button7_Click(object sender, EventArgs e)
        {
            button7.Enabled = false;
            IAgVAMCSPropagate outerOrbit = _driver.MainSequence.Insert(AgEVASegmentType.eVASegmentTypePropagate, "Outer Orbit", "-") as IAgVAMCSPropagate;
            ((IAgVAMCSSegment)outerOrbit).Properties.Color = Color.Green;
            outerOrbit.PropagatorName = "Earth Point Mass";
            ((IAgVAStoppingCondition)outerOrbit.StoppingConditions["Duration"].Properties).Trip = 86400;
            button8.Enabled = true;
        }

        private void button8_Click(object sender, EventArgs e)
        {
            this.button8.Enabled = false;
            _driver.RunMCS();
            IAgVAMCSTargetSequence startTransfer = _driver.MainSequence["Start Transfer"] as IAgVAMCSTargetSequence;
            IAgVAMCSTargetSequence finishTransfer = _driver.MainSequence["Finish Transfer"] as IAgVAMCSTargetSequence;
            IAgVAProfileDifferentialCorrector startDC = startTransfer.Profiles["Differential Corrector"] as IAgVAProfileDifferentialCorrector;
            Console.WriteLine(startDC.ControlParameters.GetControlByPaths("DV1", "ImpulsiveMnvr.Cartesian.X").FinalValue);
            IAgVAProfileDifferentialCorrector finishDC = finishTransfer.Profiles["Differential Corrector"] as IAgVAProfileDifferentialCorrector;
            Console.WriteLine(finishDC.ControlParameters.GetControlByPaths("DV2", "ImpulsiveMnvr.Cartesian.X").FinalValue);
            IAgVAMCSManeuver dv1 = startTransfer.Segments["DV1"] as IAgVAMCSManeuver;
            IAgVAManeuverImpulsive dv1Impulsive = dv1.Maneuver as IAgVAManeuverImpulsive;
            IAgVAAttitudeControlImpulsiveThrustVector dv1ThrustVector = dv1Impulsive.AttitudeControl as IAgVAAttitudeControlImpulsiveThrustVector;
            IAgCartesian cartesian = dv1ThrustVector.DeltaVVector.ConvertTo(AgEPositionType.eCartesian) as IAgCartesian;
            Console.WriteLine(cartesian.X);
            startTransfer.ApplyProfiles();
            cartesian = dv1ThrustVector.DeltaVVector.ConvertTo(AgEPositionType.eCartesian) as IAgCartesian;
            Console.WriteLine(cartesian.X);

            IAgVAMCSManeuver dv2 = finishTransfer.Segments["DV2"] as IAgVAMCSManeuver;
            IAgVAManeuverImpulsive dv2Impulsive = dv2.Maneuver as IAgVAManeuverImpulsive;
            IAgVAAttitudeControlImpulsiveThrustVector dv2ThrustVector = dv2Impulsive.AttitudeControl as IAgVAAttitudeControlImpulsiveThrustVector;
            cartesian = dv2ThrustVector.DeltaVVector.ConvertTo(AgEPositionType.eCartesian) as IAgCartesian;
            Console.WriteLine(cartesian.X);
            finishTransfer.ApplyProfiles();
            cartesian = dv2ThrustVector.DeltaVVector.ConvertTo(AgEPositionType.eCartesian) as IAgCartesian;
            Console.WriteLine(cartesian.X);
            this.button10.Enabled = true;
        }

        private void Form1_FormClosing(object sender, FormClosingEventArgs e)
        {
            if (stkRootObject != null)
            {
                stkRootObject.CloseScenario();
            }
        }

        private void button10_Click(object sender, EventArgs e)
        {
            Form2 dataForm = new Form2();
            String reportData = "";
            IAgVAMCSPropagate outerOrbit = _driver.MainSequence["Outer Orbit"] as IAgVAMCSPropagate;
            IAgDrResult result = ((IAgVAMCSSegment)outerOrbit).ExecSummary;
            Console.WriteLine((AgEDrCategories)result.Category);
            IAgDrIntervalCollection intervals = result.Value as IAgDrIntervalCollection;
            for (int i = 0; i < intervals.Count; i++)
            {
                IAgDrInterval interval = intervals[i];
                IAgDrDataSetCollection datasets = interval.DataSets;
                for (int j = 0; j < datasets.Count; j++)
                {
                    IAgDrDataSet dataset = datasets[j];
                    dataForm.FormTitle = dataset.ElementName;
                    System.Array elements = dataset.GetValues();
                    foreach (object o in elements)
                    {
                        reportData += o + "\r\n";
                    }
                }
            }
            dataForm.ReportData = reportData;
            dataForm.ShowDialog();
        }




    }
}